migrator/migrations/m_171_to_m_172_move_scope_to_collection_in_report_configurations/migration.go

```diff
@@ -215,31 +229,61 @@ func moveScopesInReportsToCollections(gormDB *gorm.DB, db *pgxpool.Pool) error {
 	accessScopeStore := accessScopePostgres.New(db)
 	collectionStore := collectionPostgres.New(db)
 
+	allConfigs := make(map[string]*storage.ReportConfiguration)
 	err := reportConfigStore.Walk(ctx, func(reportConfig *storage.ReportConfiguration) error {
-		scopeIDToConfigNames[reportConfig.GetScopeId()] = append(scopeIDToConfigNames[reportConfig.GetScopeId()], reportConfig.GetName())
+		scopeIDToConfigs[reportConfig.GetScopeId()] = append(scopeIDToConfigs[reportConfig.GetScopeId()], reportConfig)
+		allConfigs[reportConfig.GetId()] = reportConfig
 		return nil
 	})
 	if err != nil {
 		return err
 	}
 
-	for scopeID := range scopeIDToConfigNames {
-		err = createCollectionsForScope(ctx, scopeID, accessScopeStore, collectionStore)
-		if err != nil {
-			return err
+	for scopeID := range scopeIDToConfigs {
+		newScopeID, created := createCollectionsForScope(ctx, scopeID, accessScopeStore, collectionStore)
+		if created && newScopeID != scopeID {
+			// Update each of the reports with the new scope id.
+			// This is required since the scope id may have changed between RocksDB and Postgres.
+			// See migration n52_to_n53
+			configs := scopeIDToConfigs[scopeID]
+			for _, config := range configs {
+				config.ScopeId = newScopeID
+				// Do it one at a time even though it's not as performant so that at least some reports will get migrated
+				// even if any fail. It should be rare though.
+				if err := reportConfigStore.Upsert(ctx, config); err != nil {
+					log.Error(errorWithResolutionMsg(errors.Wrap(err, "Failed to update report configuration with collections to new scope"), scopeID))
+				}
+			}
 		}
 	}
 	return err
 }
 
 func errorWithResolutionMsg(err error, scopeID string) string {
+	var scopeNames []string
+	for _, config := range scopeIDToConfigs[scopeID] {
+		scopeNames = append(scopeNames, config.GetName())
+	}
 	return err.Error() + "\n" +
 		" The scope is attached to the following report configurations: " +
```

Comment:
Nit: s/is attached to/is associated with

Suggested reword of a potentially user facing message:
"Please manually create an equivalent collection and edit the listed report configurations to use this collection. Note: Reports generation for listed report configurations will not work until a collection is associated with each of them."

Reply:
@charmik-redhat